src/session.rs

```diff
@@ -508,7 +505,19 @@ impl Session {
         // here we have incoming and depacketized data before it may be dropped at buffer.push()
         let bytes_rx = data.len();
 
-        buf.push(meta, data);
+        // In RTP mode we want to retain the header. After srtp_unprotect, we need to
+        // recombine the header + the decrypted payload.
+        let data = if self.rtp_mode {
+            let hlen = meta.header.header_len;
```

Comment:
you need temp allocation here or can you do it in data directly?

Reply:
We could potentially allocate once and move the data around in the vec.

For sample based data is just the body, no header.